tests/test_model.py

```diff
@@ -1653,3 +1656,43 @@ def test_model_logp_fast_compile():
 
     with pytensor.config.change_flags(mode="FAST_COMPILE"):
         assert m.point_logps() == {"a": -1.5}
+
+
+class TestModelGraphs:
+    @staticmethod
+    def school_model(J: int) -> pm.Model:
+        y = np.array([28, 8, -3, 7, -1, 1, 18, 12])
+        sigma = np.array([15, 10, 16, 11, 9, 11, 10, 18])
+        with pm.Model(coords={"school": np.arange(J)}) as schools:
+            eta = pm.Normal("eta", 0, 1, dims="school")
+            mu = pm.Normal("mu", 0, sigma=1e6)
+            tau = pm.HalfCauchy("tau", 25)
+            theta = mu + tau * eta
+            pm.Normal("obs", theta, sigma=sigma, observed=y, dims="school")
+        return schools
+
+    @pytest.mark.parametrize(
+        argnames="var_names", argvalues=[None, ["mu", "tau"]], ids=["all", "subset"]
+    )
+    def test_graphviz_call_function(self, var_names) -> None:
+        model = self.school_model(J=8)
+        with patch("pymc.model.model_to_graphviz") as mock_model_to_graphviz:
+            model.to_graphviz(var_names=var_names)
+            mock_model_to_graphviz.assert_called_once_with(
+                model=model, var_names=var_names, formatting="plain"
+            )
+
+    def test_graphviz(self) -> None:
+        model = self.school_model(J=8)
+        g_actual = model.to_graphviz()
+        g_expected = model_to_graphviz(model)
+        assert g_actual.body == g_expected.body
+        assert g_actual.source == g_expected.source
+
+    def test_graphviz_different_models(self) -> None:
+        model_1 = self.school_model(J=8)
+        model_2 = self.school_model(J=5)
+        g_1 = model_1.to_graphviz()
+        g_2 = model_2.to_graphviz()
+        assert g_1.body != g_2.body
+        assert g_1.source != g_2.source
```

Comment:
```suggestion
            )
```

IMO the mocked test is sufficient. For the other tests to catch something that's not already caught by existing tests one would have to actively sabotage the new method

Reply:
ok! I'll do so thanks!